mnefun/_scoring.py

```diff
@@ -104,15 +104,28 @@ def extract_expyfun_events(fname, return_offsets=False):
     return these_events, resps, orig_events
 
 
-def _read_events(p, subj, ridx, raw):
+def _pick_events(events, picker):
+    old_cnt = sum(len(e) for e in events)
+    events = picker(events)
+    new_cnt = len(events)
+    print(f'  Using {new_cnt}/{old_cnt} events.')
+
+
+def _read_events(p, subj, ridx, raw, picker=None):
     ridx = np.array(ridx)
     assert ridx.ndim == 1
+    if picker == 'restrict':  # limit to events that will be processed
+        ids = p.in_numbers
+        picker = None
+        print('Events restricted to those in params.in_numbers')
+    else:
+        ids = None
     events = list()
     for fname in get_event_fnames(p, subj, ridx):
-        these_events = read_events(fname)
-        if len(np.unique(these_events[:, 0])) != len(these_events):
-            raise RuntimeError('Non-unique event samples found in %s'
-                               % (fname,))
+        these_events = read_events(fname, include=ids)
+        # if len(np.unique(these_events[:, 0])) != len(these_events):
+        #     raise RuntimeError('Non-unique event samples found in %s'
+        #                        % (fname,))
```

Comment:
Rather than commenting this out, it seems like this should just be moved down after the `events = _pick_events(...)` conditional below

Reply:
That's fine - I'll make that change.  I just wasn't sure it is necessary at all if, just a few lines below, the concatenated events are checked for unique times.

Reply:
ohhh that happens already. Then yes this is not necessary, just delete these lines rather than commenting them

Reply:
Actually, I recall now that the reason I didn't keep the time check within the fname loop is in case picker IS a function and the read_events() line crashes when include=None.  Can we leave it out for now and assume the post-concatenation check is sufficient?

Reply:
Ah, sorry, I was in the middle of typing when our meeting started.  I'll delete the lines and repush.